app/kuma-dp/pkg/dataplane/envoy/envoy.go

```diff
@@ -156,3 +167,27 @@ func (e *Envoy) Start(stop <-chan struct{}) error {
 		return err
 	}
 }
+
+func (e *Envoy) version() (*EnvoyVersion, error) {
+	binaryPathConfig := e.opts.Config.DataplaneRuntime.BinaryPath
+	resolvedPath, err := lookupEnvoyPath(binaryPathConfig)
+	if err != nil {
+		return nil, err
+	}
+	arg := "--version"
+	command := exec.Command(resolvedPath, arg)
+	runLog.Info("getting version of Envoy")
+	output, err := command.Output()
+	if err != nil {
+		runLog.Error(err, fmt.Sprintf("the envoy excutable was found at %s but an error occurred when executing it with arg %s", resolvedPath, arg))
+		return nil, err
```

Comment:
In Go, you either log or return an error. The only exception for our codebase is `run.go` of binaries. Here - wrap an error and return it

Reply:
Thank you, I fixed it.